### PR TITLE
Added appveyor script

### LINKED ISSUE #1218: Set up Windows automated builds on Appveyor
Like travis but for windows

### PR DESCRIPTION
**Pull request by ctrlaltca**
It will test msvc builds and create windows packages for both 32 and 64 bits. Fix #1218.
If you're compiling windows packages (@woogerboy21), you could need to review these changes:
 * Forced /MD flag to the visual studio compiler (fixes linking)
 * removed the hardcoded "win32" text in the package name

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    - c:\protobuf
+    - c:\protoc
+    - c:\zlib
+environment:
+    matrix:
+        - vc_arch: amd64
+          choco_arch:
+          nuget_arch: x64
+          target_arch: x86_64
+          qt_ver: 5.5\msvc2013_64
```
Comment (Daenyth) [inline appveyor.yml]: Don't we want 5.4?

Is it possible for us to also test Qt4?

Comment (ctrlaltca) [inline appveyor.yml]: For windows 5.5 is fine; 5.5 is problematic only for osx (rpath enabled by default, problems in detecting clang features).
Appveyor currently only supports qt 5.3, 5.4 and 5.5. We could download and install qt4 manually, but that would take a lot of time.

Comment (woogerboy21) [inline appveyor.yml]: @Daenyth I've been using 5.5 for windows clients for some time now

**Comment by woogerboy21** [comment]
@ctrlaltca Are you saving the builds off any were?

**Comment by ctrlaltca** [comment]
Not yet; it's planned as step 2.

**Comment by woogerboy21** [comment]
Is there a reason you dropped the win32 as well?  To me its pretty obvious which binary goes to what what OS but one thing you may want to add is the QT version some were in the binary name.  Otherwise you will end up with name collisions when creating both Qt4 & Qt5 packages.

**Comment by ctrlaltca** [comment]
I dropped the hardcoded win32 because that would end up also in the name of x86_64 packages.
Adding the qt version is a nice idea.

**Comment by woogerboy21** [comment]
I see you pull the proto files if dont exist :wink:

**Comment by ctrlaltca** [comment]
last win32 package: https://ci.appveyor.com/project/ctrlaltca/cockatrice/build/job/jvbbvghk75xo3ibn/artifacts
last win64 package: 
https://ci.appveyor.com/project/ctrlaltca/cockatrice/build/job/ibkgcbssfmbgxmla/artifacts

**Comment by woogerboy21** [comment]
Those are only Qt5 builds though correct?

**Comment by woogerboy21** [comment]
@ctrlaltca Just curious, is this the script that will be used to test each PR put in?  And if so are we really wanting a binary saved each time?

**Comment by ctrlaltca** [comment]
> @ctrlaltca Just curious, is this the script that will be used to test each PR put in? And if so are we really wanting a binary saved each time?

No, i just started working on it but forgot to do it on another branch; fixed

**Comment by woogerboy21** [comment]
Is this safe to remove the PS content near the end that manipulates the final binary and use for testing windows compiling results?  I for one would be interested in seeing this implemented as a first go around to allow a better testing for windows compiling and later add the binary file functionality.

**Comment by ctrlaltca** [comment]
The last PS line is just renaming the resulting exe, adding the arch it has been built for, and telling appveyor what files to conserve before trashing the worker vm. IMHO this PR is quite done.

**Comment by woogerboy21** [comment]
:+1:   Wasnt sure if you had wanted to add more to the PS section.  But by your statement it does not appear to be the case for now.

**Comment by woogerboy21** [comment]
Oh one other things, is there some were I could glance at the output of the run for a build in appveyor?  The links you provided earlier dont seem to show anything any longer.

**Comment by ZeldaZach** [comment]
I'm assuming this is good to go then

**Comment by woogerboy21** [comment]
Looks ok to me.

**Comment by ctrlaltca** [comment]
As per #1562, having received two signoffs, i'm gonna merge this.

**Pull request merged**